### ISSUE TITLE
Rude edits not detected when changing (primary) constructor initializer

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#68731]
See disabled tests.

1)
```C#
class C { C() : base(stackalloc int[1]) {} C(Span<int> span) {} }
```

```C#
class C { C() : base(stackalloc int[2]) {} C(Span<int> span) {} }
```

2)

```C#
class C { C(int x) : this(() => x) {} C(Func<int> f) {} }
```

```C#
class C { C(int x) : this(() => 1) {} C(Func<int> f) {} }
```

etc.
